xarray/tests/test_backends_datatree.py

```diff
@@ -73,111 +156,44 @@ def test_netcdf_encoding(self, tmpdir, simple_datatree):
-
-            root_group.createDimension("lat", 1)
-            root_group.createDimension("lon", 2)
-            root_group.createVariable("root_variable", np.float64, ("lat", "lon"))
-
-            group_1_var = group_1.createVariable(
-                "group_1_var", np.float64, ("lat", "lon")
-            )
-            group_1_var[:] = np.array([[0.1, 0.2]])
-            group_1_var.units = "K"
-            group_1_var.long_name = "air_temperature"
-
-            subgroup_1.createDimension("lat", 2)
-
-            subgroup1_var = subgroup_1.createVariable(
-                "subgroup1_var", np.float64, ("lat", "lon")
-            )
-            subgroup1_var[:] = np.array([[0.1, 0.2]])
-        with pytest.raises(ValueError):
-            open_datatree(filepath)
-
-    def test_open_groups(self, tmpdir) -> None:
-        """Test `open_groups` with netCDF4 file with the same unaligned structure:
-        Group: /
-        │   Dimensions:        (lat: 1, lon: 2)
-        │   Dimensions without coordinates: lat, lon
-        │   Data variables:
-        │       root_variable  (lat, lon) float64 16B ...
-        └── Group: /Group1
-            │   Dimensions:      (lat: 1, lon: 2)
-            │   Dimensions without coordinates: lat, lon
-            │   Data variables:
-            │       group_1_var  (lat, lon) float64 16B ...
-            └── Group: /Group1/subgroup1
-                    Dimensions:        (lat: 2, lon: 2)
-                    Dimensions without coordinates: lat, lon
-                    Data variables:
-                        subgroup1_var  (lat, lon) float64 32B ...
-        """
-        filepath = tmpdir + "/unaligned_subgroups.nc"
-        with nc4.Dataset(filepath, "w", format="NETCDF4") as root_group:
-            group_1 = root_group.createGroup("/Group1")
-            subgroup_1 = group_1.createGroup("/subgroup1")
-
-            root_group.createDimension("lat", 1)
-            root_group.createDimension("lon", 2)
-            root_group.createVariable("root_variable", np.float64, ("lat", "lon"))
+    def test_open_datatree(self, unaligned_datatree_nc) -> None:
+        """Test if `open_datatree` fails to open a netCDF4 with an unaligned group hierarchy."""
 
-            group_1_var = group_1.createVariable(
-                "group_1_var", np.float64, ("lat", "lon")
-            )
-            group_1_var[:] = np.array([[0.1, 0.2]])
-            group_1_var.units = "K"
-            group_1_var.long_name = "air_temperature"
-
-            subgroup_1.createDimension("lat", 2)
-
-            subgroup1_var = subgroup_1.createVariable(
-                "subgroup1_var", np.float64, ("lat", "lon")
-            )
-            subgroup1_var[:] = np.array([[0.1, 0.2]])
+        with pytest.raises(
+            ValueError,
+            match=(
+                re.escape(
+                    "group '/Group1/subgroup1' is not aligned with its parents:\nGroup:\n"
+                )
```

Comment:
I wonder if this could be just an `r` string? Something like
```suggestion
                r"group '/Group1/subgroup1' is not aligned with its parents:[\n]Group:[\n].*"
```